crates/fluvio/src/producer/mod.rs

```diff
@@ -196,354 +486,69 @@ impl TopicProducer {
     }
 
-    #[test]
-    fn test_assemble_requests() {
-        let partitions_by_spu = {
-            let mut pbs = HashMap::new();
-
-            let spu_0 = {
-                let mut s0_partitions = HashMap::new();
-                let partition_0_records = vec![Record::new("A"), Record::new("B")];
-                let partition_1_records = vec![Record::new("C"), Record::new("D")];
-                s0_partitions.insert(0, partition_0_records);
-                s0_partitions.insert(1, partition_1_records);
-                s0_partitions
-            };
-
-            let spu_1 = {
-                let mut s1_partitions = HashMap::new();
-                let partition_2_records = vec![Record::new("E"), Record::new("F")];
-                let partition_3_records = vec![Record::new("G"), Record::new("H")];
-                s1_partitions.insert(2, partition_2_records);
-                s1_partitions.insert(3, partition_3_records);
-                s1_partitions
-            };
-            pbs.insert(0, spu_0);
-            pbs.insert(1, spu_1);
-            pbs
-        };
-
-        let requests = assemble_requests("TOPIC", partitions_by_spu);
-        assert_eq!(requests.len(), 2);
-
-        // SPU 0
-        {
-            let (spu0, request) = requests.iter().find(|(spu, _)| *spu == 0).unwrap();
-            assert_eq!(*spu0, 0);
-            assert_eq!(request.topics.len(), 1);
-            let topic_request = request.topics.get(0).unwrap();
-            assert_eq!(topic_request.name, "TOPIC");
-            assert_eq!(topic_request.partitions.len(), 2);
-
-            let partition_0_request = topic_request
-                .partitions
-                .iter()
-                .find(|p| p.partition_index == 0)
-                .unwrap();
-            assert_eq!(partition_0_request.records.batches.len(), 1);
-            let batch = partition_0_request.records.batches.get(0).unwrap();
-            assert_eq!(batch.records().len(), 2);
-            let record_0_0 = batch.records().get(0).unwrap();
-            assert_eq!(record_0_0.value.as_ref(), b"A");
-            let record_0_1 = batch.records().get(1).unwrap();
-            assert_eq!(record_0_1.value.as_ref(), b"B");
-
-            let partition_1_request = topic_request
-                .partitions
-                .iter()
-                .find(|p| p.partition_index == 1)
-                .unwrap();
-            assert_eq!(partition_1_request.records.batches.len(), 1);
-            let batch = partition_1_request.records.batches.get(0).unwrap();
-            assert_eq!(batch.records().len(), 2);
-            let record_1_0 = batch.records().get(0).unwrap();
-            assert_eq!(record_1_0.value.as_ref(), b"C");
-            let record_1_1 = batch.records().get(1).unwrap();
-            assert_eq!(record_1_1.value.as_ref(), b"D");
-        }
+    /// Wait for all record metadata of all records sent using smartmodule
+    #[cfg(feature = "smartengine")]
```

Comment:
why this needs to be under smartengine`?

Reply:
Only when we are using smartengine we can have multiple records from a single send call